src/Raven.Studio/typescript/models/database/settings/databaseSettingsModels.ts

```diff
@@ -0,0 +1,444 @@
+﻿/// <reference path="../../../../typings/tsd.d.ts"/>
+import jsonUtil = require("common/jsonUtil");
+import getFolderPathOptionsCommand = require("commands/resources/getFolderPathOptionsCommand");
+import genUtils = require("common/generalUtils");
+
+type configurationOrigin = "Default" | "Server" | "Database";
+
+export abstract class settingsEntry {
+
+    data: Raven.Server.Config.ConfigurationEntryDatabaseValue;
+
+    keyName = ko.observable<string>();
+    showEntry = ko.observable<boolean>();
+
+    isServerWideOnlyEntry = ko.observable<boolean>();
+    serverOrDefaultValue: KnockoutComputed<string>;
+    hasServerValue: KnockoutComputed<boolean>;
+    
+    effectiveValue: KnockoutComputed<string>;
+    effectiveValueOrigin: KnockoutComputed<configurationOrigin>;
+    
+    rawValue = ko.observable<string>();
+    hasPendingContent = ko.observable<boolean>(false);
+    
+    // These 2 are needed for Summary view
+    pendingValueText: KnockoutComputed<string>;
+    effectiveValueInUseText: KnockoutComputed<string>;
+
+    entryClassForSummaryMode: KnockoutComputed<string>;
+    descriptionHtml: KnockoutComputed<string>;
+
+    protected constructor(data: Raven.Server.Config.ConfigurationEntryDatabaseValue) {
+        this.data = data;
+
+        this.keyName(this.data.Metadata.Keys[this.data.Metadata.Keys.length - 1]);
+
+        this.serverOrDefaultValue = ko.pureComputed(() => _.isEmpty(this.data.ServerValues) ? this.data.Metadata.DefaultValue : this.data.ServerValues[this.keyName()].Value);
+        this.hasServerValue = ko.pureComputed(() => !_.isEmpty(this.data.ServerValues));
+        
+        this.descriptionHtml = ko.pureComputed(() => {
+            const rawDescription = data.Metadata.Description;
+            return rawDescription ? 
+                `<div><span>${genUtils.escapeHtml(rawDescription)}</span></div>` :
+                `<div><span class="text-muted">No description is available</span></div>`;
```

Comment:
do we need span ? We can put text-muted on div?

Reply:
removed